### PR TITLE
Add SDK flag for retaining undefined properties

### PR DESCRIPTION
**Pull request by alexd-codaio**
@jonathan-codaio @coda/packs PTAL

Will be useful as a backstop in cases where we have dynamic schema coming back from normal formulas/actions

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ This changelog keeps track of all changes to the packs-sdk. We follow convention
 
 - Added "force" option to `LinkSchema` for `LinkDisplayType.Embed`.
 - Added "downloadFilename" option to `storeBlob` and `storeUrl` in `TemporaryBlobStorage` to specify the file name that it should download as when opening the temporary url.
+- Added "downloadFilenameincludeUndefinedProperties" option to `makeObjectscheam` that will retain properties on objects that are not defined within the object's schema.
```
[the remark this answers is not in the document]
Comment (alexd-codaio) [reply CHANGELOG.md]: Done.

File: schema.ts [hunk 2]
```diff
@@ -807,6 +807,16 @@ export interface ObjectSchemaDefinition<K extends string, L extends string> exte
    * See {@link makeAttributionNode}.
    */
   attribution?: AttributionNode[];
+  /**
+   * Specifies that object instances with this schema can contain additional properties not defined
+   * in the schema, and that the packs infrastructure should retain these undefined properties
+   * rather than stripping them.
+   * 
+   * Properties not declared in the schema will not work properly in Coda: the cannot be
```
[the remark this answers is not in the document]
Comment (alexd-codaio) [reply schema.ts]: Done.

File: schema.ts [hunk 3]
```diff
@@ -807,6 +807,16 @@ export interface ObjectSchemaDefinition<K extends string, L extends string> exte
...
+   * use natively in the formula language and will not have correct types in Coda. But, in certain
```
[the remark this answers is not in the document]
Comment (alexd-codaio) [reply schema.ts]: Done.

File: schema.ts [hunk 4]
```diff
@@ -807,6 +807,16 @@ export interface ObjectSchemaDefinition<K extends string, L extends string> exte
...
+  /**
+   * Specifies that object instances with this schema can contain additional properties not defined
+   * in the schema, and that the packs infrastructure should retain these undefined properties
+   * rather than stripping them.
+   * 
+   * Properties not declared in the schema will not work properly in Coda: the cannot be
+   * use natively in the formula language and will not have correct types in Coda. But, in certain 
+   * scenarios they can be useful.
+   */
+  includeUndefinedProperties?: boolean;
```
[the remark this answers is not in the document]
Comment (alexd-codaio) [reply schema.ts]: Agreed. Done.

[the remark this answers is not in the document]
Comment (ekoleda-codaio) [reply schema.ts]: I'd probably recommend something like "unknown", the key being that they exist but we don't know what they are. Even "unspecified" still makes me think the value isn't set, which is different.

**Comment by ekoleda-codaio** [comment]
> Will be useful as a backstop in cases where we have dynamic schema coming back from normal formulas/actions

Am I right in understanding that this would work for static sync tables with dynamic schemas, but not for dynamic sync tables?

**Comment by alexd-codaio** [comment]
> > Will be useful as a backstop in cases where we have dynamic schema coming back from normal formulas/actions
> 
> Am I right in understanding that this would work for static sync tables with dynamic schemas, but not for dynamic sync tables?

This specific change isn't really related to sync tables at all per se, since we have all the mechanisms to let the caller return the correct schema...  static + dynamic or fully dynamic.

I'm adding this (back) specifically since this will help the sync table + action button in the table scenario, where the shape of the schema on the response doesn't really matter (only the identity) but without this field we're deleting these unknown keys.

File: schema.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexd-codaio) [reply schema.ts]: Done.

**Review by jonathan-codaio — APPROVED** [review]

**Pull request merged**